gm4_balloon_animals/data/gm4_balloon_animals/functions/wandering_trader/animal_one/random_calc.mcfunction

```diff
@@ -0,0 +1,17 @@
+# Gets 2 random numbers to choose between common or rare animal and which animal in those groups to put for trade
+# @s = marker
+# at @s, located at the trader
+# from gm4_balloon_animals:wandering_trader/pick_two_animals
+
+# check 30-70% chance of common-rare animal
+execute store result score $a1_rare_calc gm4_balloon_animals_data run data get entity @s UUID[0]
+scoreboard players operation $a1_rare_calc gm4_balloon_animals_data %= #10 gm4_balloon_animals_data
+
+
+execute store result score $a1_choice gm4_balloon_animals_data run data get entity @s UUID[3]
+
+execute if score $a1_rare_calc gm4_balloon_animals_data matches 0..2 run scoreboard players operation $a1_choice gm4_balloon_animals_data %= #10 gm4_balloon_animals_data
+execute if score $a1_rare_calc gm4_balloon_animals_data matches 3..9 run scoreboard players operation $a1_choice gm4_balloon_animals_data %= #8 gm4_balloon_animals_data
```

Comment:
I'd recommend using
1. a (or multiple) computational loot table(s) here to determine the animal type instead of the UUID. That way you don't need a temporary marker and you have greater control over the way the number is rolled. OR
2. the `/random` command (will require 1.20.2 or above).

Reply:
Use a computational loot table then!